seed/static/seed/js/controllers/data_quality_admin_controller.js

```diff
@@ -231,16 +276,46 @@ angular.module('BE.seed.controller.data_quality_admin', [])
         spinner_utility.show();
         data_quality_service.save_data_quality_rules($scope.org.org_id, rules).then(function (rules) {
           loadRules(rules);
-          $scope.rules_updated = true;
           modified_service.resetModified();
         }).then(function (data) {
           $scope.$emit('app_success', data);
         }).catch(function (data) {
           $scope.$emit('app_error', data);
         }).finally(function () {
+          $scope.rules_updated = true;
           spinner_utility.hide();
         });
       };
+      $scope.change_condition = function (rule) {
+        $scope.rules_updated = false;
+        $scope.defaults_restored = false;
+        $scope.rules_reset = false;
+        if (!rule.condition) rule.condition = null;
+        if (rule.condition === 'include' || rule.condition === 'exclude' && rule.data_type !== 'string') rule.data_type = 'string';
+        if (_.isMatch(rule, {condition: 'range', data_type: 'string'})) rule.data_type = null;
+        if (_.isMatch(rule, {condition: 'not_null', data_type: 'string'})) rule.text_match = null;
+
+        var condition = rule.condition;
+        if (condition === 'required') {
+          rule.required = false;
+          $scope.change_required(rule);
+        } else if (condition === 'not_null') {
+          rule.required = false;
+          rule.not_null = rule.required;
+          $scope.change_not_null(rule);
+        }
+        if (rule.condition !== 'range') {
+          rule.units = '';
+          rule.min = null;
+          rule.max = null;
+        }
```

Comment:
Should we do something similar for text_match and include/exclude?

Reply:
This is done about 11 lines up:
![Screen Shot 2020-05-06 at 2 21 29 PM](https://user-images.githubusercontent.com/49968203/81225326-4bf65780-8fa6-11ea-95c2-26a3dca965df.png)